share/pot/clone.sh

```diff
@@ -80,6 +80,7 @@ _cj_zfs()
 		rm -f "$_pdir/conf/fscomp.conf"
 	fi
 	_debug "Cloning $_potbase with snap $_snap"
+	_update_fscomp "$_potbase"
```

Comment:
The one thing I'm not so sure about is that it seems like the clone command now "fixes" the existing potbase. From a user perspective, I would be surprised that cloning an image actually changes the image it is cloning from.

Reply:
IIRC, this is to ensure that `clone`with the new layout is automatically updating the old one. It's only a configuration file, tho.
It's in the spirit of `update-config`, to ensure that configuration files evolve consistently when introducing a breaking change.
do you have a better approach? I'm not super happy with that, but it was the best thing I had in mind

Reply:
What about the following logic:

Replace

```
cp "${_pbdir}/conf/fscomp.conf" "$_pdir/conf/fscomp.conf"
```

With something like:

```
< "${_pbdir}/conf/fscomp.conf"  sed "s|^${POT_FS_ROOT}/jails/$_pname/m|g" >"$_pdir/conf/fscomp.conf"
```

And only actually clean up the fscomp.conf file whenever `update-config`, `mount-in` or `mount-out` are called.
(in which case the check if it's an old format won't be necessary, as you could simply always apply the filtering sed command from above to a temp file, append/remove the  mountpoint if necessary and then move the new file into place)

Reply:
I like your approach. For `clone` is indeed not intuitive to change the base.
I'll follow your suggestion and changing paths on the fly

Reply:
@pizzamig Are you currently working on this, or should I give it a shot?